DEVELOPMENT.md

```diff
@@ -119,6 +119,14 @@ Disable the following inspections:
 - ``Java | Performance | Call to 'Arrays.asList()' with too few arguments``,
 - ``Java | Abstraction issues | 'Optional' used as field or parameter type``.
 
+Because we work a lot with SQL, IntelliJ IDEA's language injection feature is useful in the project. To make the injection automatic (and permanent), certain code elements can be annotated with the `@org.intellij.lang.annotations.Language` annotation, and you'll encounter it in many places in Trino code. Before it's actually useful, though, there are three related configuration that you may have to change:
+
+- Set ``Languages & Frameworks | SQL Dialects``: By default IntelliJ will highlight all SQL-injected values in yellow, saying that the SQL dialect is not configured. You can disable the inspection, but it's better to set the SQL Dialect globally using this option. There's no `Trino` dialect, unfortunately, but the `Generic SQL` one is close enough. You can vary the dialect for specific files, for example set the PostgreSQL dialect in PostgreSQL connector tests.
+- Disable inspection ``SQL | No data source configured``: Language injection for SQL is made significantly more useful when it is also aware of the structure of the database; unfortunately in Trino we don't have a single data source - and the tests may even be using a completely made-up database structure - so this inspection is not very helpful.
+- Inspection ``Language injection | Language mismatch``: This inspection will highlight any local variable which is used in a context annotated with language injection, but is not annotated itself. We don't recommend disabling it outright, because many people find it useful, but we're also not keen on over-using the annotations where the language can be perfectly well inferred from context (e.g. when the variable is used as an argument for a parameter annotated with specific language).
+
+Considering the above, we recommend that you annotate all API parameters which are expecting to take a `String` containing an SQL statement (or any other language, like regular expressions). You may annotate also any local variables in new code, but refrain from doing it in existing code, as it only generates noise in the commit history.
+
```

Comment:
This is very verbose. In order to understand what it is all about a reader is required to spend noticeable amount of energy.  This concerns me as I am afraid people won't read it.

How about:
```
### Language injection

To use language injection certain code elements can be annotated with the 
`@org.intellij.lang.annotations.Language` annotation. 

To make it useful we recommend:
- Set ``Languages & Frameworks | SQL Dialects``
- Disable inspection ``SQL | No data source configured``:
- Disable inspection ``Language injection | Language mismatch``.

Considering the above, we recommend to annotate with `@Language`:
 -  all API parameters which are expecting to take a `String` 
 containing an SQL statement (or any other language, like regular expressions). 
  - local variables which otherwise would no be properly recognized 
  by IDE  as language injection.
```

Reply:
I would not worry too much about people not reading it. I would expect that if someone reads DEVELOPMENT.md at all, they will over go through this (not that I expect that many people would any part of that file :) ).

Still I like the shorter version